packages/service-library/tests/fastapi/test_request_decorators.py

```diff
@@ -2,121 +2,69 @@
-        print("server is up and waiting for requests...")
-        yield ServerInfo(url, proc)
-        print("server is closing...")
-        proc.terminate()
-        print("server terminated")
-
-
-def test_cancel_on_disconnect(get_unused_port: Callable[[], int]):
-
-    with server_lifetime(port=get_unused_port()) as server:
-        url, proc = server
-        print("--> testing server")
-        response = requests.get(f"{server.url}/example?wait=0", timeout=2)
-        print(response.url, "->", response.text)
-        response.raise_for_status()
-        print("<-- server responds")
-
-        print("--> testing server correctly cancels")
-        with pytest.raises(requests.exceptions.ReadTimeout):
-            response = requests.get(f"{server.url}/example?wait=2", timeout=0.5)
-        print("<-- testing server correctly cancels done")
-
-        print("--> testing server again")
-        # NOTE: the timeout here appears to be sensitive. if it is set <5 the test hangs from time to time
-        response = requests.get(f"{server.url}/example?wait=1", timeout=5)
-        print(response.url, "->", response.text)
-        response.raise_for_status()
-        print("<-- testing server again done")
-
-        # kill service
-        server.proc.terminate()
-        assert server.proc.stdout
-        server_log = server.proc.stdout.read().decode("utf-8")
-        print(
-            f"{server.url=} stdout",
-            "-" * 10,
-            "\n",
-            server_log,
-            "-" * 30,
-        )
-        # server.url=http://127.0.0.1:44077 stdout ----------
-        # Sleeping for 0.00
-        # Sleep not cancelled
-        # INFO:     127.0.0.1:35114 - "GET /example?wait=0 HTTP/1.1" 200 OK
-        # Sleeping for 2.00
-        # Exiting on cancellation
-        # Sleeping for 1.00
-        # Sleep not cancelled
-        # INFO:     127.0.0.1:35134 - "GET /example?wait=1 HTTP/1.1" 200 OK
-
-        assert MESSAGE_ON_HANDLER_CANCELLATION in server_log
+    """
+    Tests that the decorator's wrapper waits for the poller task to finish
+    its cleanup, even if the handler finishes first, without needing a full server.
+    """
+    long_running_poller_mock.called = False
+    handler_was_called = False
+
+    @cancel_on_disconnect
+    async def my_handler(request: Request):
+        nonlocal handler_was_called
+        handler_was_called = True
+        await asyncio.sleep(0.1)  # Simulate quick work
+        return "Success"
+
+    # Mock a fastapi.Request object
+    mock_request = AsyncMock(spec=Request)
+    mock_request.is_disconnected.return_value = False
+
+    # ---
+    tasks_before = asyncio.all_tasks()
+
+    # Call the decorated handler
+    _ = await my_handler(mock_request)
+
+    tasks_after = asyncio.all_tasks()
+    # ---
+
+    assert handler_was_called
+    assert long_running_poller_mock.called == True
+
+    # Check that no background tasks were left orphaned
+    assert tasks_before == tasks_after
```

Comment:
Why dont you leave the former example as well? The nuew implementation should pass also the old test.